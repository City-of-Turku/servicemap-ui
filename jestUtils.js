import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });