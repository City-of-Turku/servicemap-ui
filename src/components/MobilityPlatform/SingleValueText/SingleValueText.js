import React from 'react';
import { Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import useLocaleText from '../../../utils/useLocaleText';

/** Renders text that includes react-intl translation and object containing localized values */

const SingleValueText = ({
  classes, intl, messageId, textObj,
}) => {
  const getLocaleText = useLocaleText();

  return (
    <div className={classes.margin}>
      <Typography component="p" variant="body2">
        {intl.formatMessage({ id: messageId }, { value: getLocaleText(textObj) })}
      </Typography>
    </div>
  );
};

SingleValueText.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  messageId: PropTypes.string,
  textObj: PropTypes.objectOf(PropTypes.any),
};

SingleValueText.defaultProps = {
  messageId: '',
  textObj: {},
};

export default SingleValueText;
