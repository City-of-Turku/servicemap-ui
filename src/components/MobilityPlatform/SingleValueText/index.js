import { withStyles } from '@material-ui/core';
import { injectIntl } from 'react-intl';
import SingleValueText from './SingleValueText';
import styles from './styles';

export default withStyles(styles)(injectIntl(SingleValueText));
