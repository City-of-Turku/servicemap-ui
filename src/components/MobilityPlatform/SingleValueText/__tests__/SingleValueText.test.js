// Link.react.test.js
import React from 'react';
import SingleValueText from '../index';
import { initialState } from '../../../../redux/reducers/user';
import { getRenderWithProviders } from '../../../../../jestUtils';

const mockProps = {
  messageId: 'mobilityPlatform.content.address',
  textObj: {
    address_fi: 'Testikatu',
    address_en: 'Test street',
    address_sv: 'Test gata',
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<SingleValueText />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<SingleValueText {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<SingleValueText {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(`Osoite: ${mockProps.textObj.address_fi}`);
  });
});
