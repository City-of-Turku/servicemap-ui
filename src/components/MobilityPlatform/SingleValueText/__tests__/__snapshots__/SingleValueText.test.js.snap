// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleValueText /> should match snapshot 1`] = `
<div>
  <div
    class="injectIntl(SingleValueText)-margin-1"
  >
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      Osoite: Testikatu
    </p>
  </div>
</div>
`;
